src/libraries/System.Collections/src/System/Collections/Generic/Queue.cs

```diff
@@ -386,6 +386,64 @@ public void TrimExcess()
             }
         }
 
+        /// <summary>
+        /// Ensures that the capacity of this Queue is at least the specified <paramref name="capacity"/>.
+        /// </summary>
+        /// <param name="capacity">The minimum capacity to ensure</param>
+        public int EnsureCapacity(int capacity)
+        {
+            if (capacity < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(capacity), capacity, SR.ArgumentOutOfRange_NeedNonNegNum);
+            }
+
+            if (_array.Length < capacity)
+            {
+                int newCapacity = (int)((long)(_array.Length == 0 ? 1 : _array.Length) * (long)GrowFactor / 100);
+                while (newCapacity < capacity)
```

Comment:
Similar to the list implementation, this should account for arithmetic overflow.

Reply:
Fixed: because GrowFactor is a config (not always 200) so here uses another way to handle overflow.